qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -110,93 +118,245 @@ def __init__(self, data, validate=True):
+            if validate and not Clifford._is_symplectic(self.tableau()):
                 raise QiskitError(
-                    "Invalid Clifford. Input StabilizerTable is not a valid symplectic matrix."
+                    "Invalid Clifford. Input PauliList is not a valid symplectic matrix."
                 )
 
         # Initialize BaseOperator
-        super().__init__(num_qubits=self._table.num_qubits)
+        super().__init__(num_qubits=self._paulis.num_qubits)
 
     def __repr__(self):
-        return f"Clifford({repr(self.table)})"
+        return f"Clifford({repr(self.paulis)})"
 
     def __str__(self):
-        return "Clifford: Stabilizer = {}, Destabilizer = {}".format(
-            str(self.stabilizer.to_labels()), str(self.destabilizer.to_labels())
-        )
+        labels = self.paulis.to_labels()
+        num_qubits = self.num_qubits
+        stabilizer = slice(num_qubits, 2 * num_qubits)
+        destabilizer = slice(0, num_qubits)
+        return f"Clifford: Stabilizer = {labels[stabilizer]}, Destabilizer = {labels[destabilizer]}"
 
     def __eq__(self, other):
         """Check if two Clifford tables are equal"""
-        return super().__eq__(other) and self._table == other._table
+        return super().__eq__(other) and self._paulis == other._paulis
 
     # ---------------------------------------------------------------------
     # Attributes
     # ---------------------------------------------------------------------
     def __getitem__(self, key):
         """Return a stabilizer Pauli row"""
-        return self._table.__getitem__(key)
+        return self._paulis.__getitem__(key)
 
     def __setitem__(self, key, value):
         """Set a stabilizer Pauli row"""
-        self._table.__setitem__(key, value)
+        self._paulis.__setitem__(key, value)
 
     @property
+    def paulis(self):
+        """Return PauliList"""
+        return self._paulis
+
+    @paulis.setter
+    def paulis(self, value):
+        """Set the Pauli list"""
+        if not isinstance(value, PauliList):
+            value = PauliList(value)
+        self._paulis._z[:, :] = value._paulis._z
+        self._paulis._x[:, :] = value._paulis._x
+        self._paulis._phase[:] = value._paulis._phase
+
+    @property
+    @deprecate_function(
+        "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the releasedate. "
+        "Use Clifford.paulis method instead.",  # pylint:disable=bad-docstring-quotes
+    )
     def table(self):
         """Return StabilizerTable"""
-        return self._table
+        return StabilizerTable(self.tableau(), phase=self.paulis.phase // 2)
 
     @table.setter
+    @deprecate_function(
+        "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+        "and will be removed no sooner than 3 months after the releasedate. "
+        "Use Clifford.paulis method instead.",  # pylint:disable=bad-docstring-quotes
```

Comment:
This is crazy - it's actually a Python/pylint bug that it throws this error with no docstring! https://github.com/PyCQA/pylint/issues/3077.

Reply:
Yes. This is workaround...

Reply:
Yeah, that's really unfortunate!  Good find - this is definitely the correct way to write the functions.